src/editor.c

```diff
@@ -2045,8 +2114,13 @@ gboolean editor_show_calltip(GeanyEditor *editor, gint pos)
 	editor_find_current_word(editor, pos - 1, word, sizeof word, NULL);
 	if (word[0] == '\0')
 		return FALSE;
-
-	str = find_calltip(word, editor->document->file_type);
+	
+	scope[0] = '\0';
+	//const gchar *sname;
+	//symbols_get_current_scope(editor->document, &sname);
+	//printf("Current scope: %s\n", sname);
+	read_scope_prefix(editor, pos - 1, scope, sizeof scope, NULL);
+	str = find_calltip(word, scope, editor->document->file_type);
```

Comment:
I would have liked to use symbols_get_current_scope here rather than write my own read_scope_prefix function, but symbols_get_current_scope kept returning "unknown" for some reason. if anyone knows how to make that function work please do